Zend/zend_API.c

```diff
@@ -539,7 +555,7 @@ static const char *zend_parse_arg_impl(int arg_num, zval *arg, va_list *va, cons
 					break;
 				} else {
 					if (is_callable_error) {
-						*severity = E_WARNING;
+						*severity = strict ? (E_EXCEPTION | E_ERROR) : E_WARNING;
```

Comment:
Correct. However, `zend_parse_arg_impl()` is only called by [`zend_parse_arg()`](https://github.com/dstogov/php-src/blob/scalar_type_hints_v5/Zend/zend_API.c#L595) which would ignore the `severity` parameter in this case **unless** in quiet mode. And if it's in quiet mode, simply setting it to anything other than `E_STRICT` would suffice.

Though this branch is in the error path, so perhaps it's not critical and the readability gain is better than the potential branch savings.